### ISSUE TITLE
The companion goes into "standby"

### ISSUE THREAD

**Issue by Federico-Catelotti** [bitfocus/companion-module-studiocoast-vmix#159]
Often the companion is as if it goes into standby. I have to press the keys a few times and then it reconnects.

**Comment by Federico-Catelotti** [comment]
it's like API pooling doesn't work. The increase also changes nothing.

**Comment by nanus** [comment]
Hello I think I have the same problem: 
I connect to two vmix instances on AWS cloud and in connection panel status is green but buttons become unresponsive after a while and after few key presses or web button presses there is a connection reset.

I've also disabled API polling interval but it does not resolve.
I did not have any trouble with companion v.2.1.4.

Here is a short log:
22.03.21 17:44:12 instance(cloudvideo1): Connected Function Socket
22.03.21 17:44:10 instance(cloudvideo3): Function Socket err: read ECONNRESET
22.03.21 17:44:10 instance(cloudvideo1): Function Socket err: read ECONNRESET
22.03.21 17:44:10 device(AL43J2C05812): Button 1.2 released
22.03.21 17:43:32 instance(vmix-asus): Command Response: FUNCTION OK Completed,
22.03.21 17:43:32 device(AL43J2C05812): Button 1.2 pressed
22.03.21 17:43:32 instance(vmix-asus): Sending command: FUNCTION CutDirect input=1&mix=0
22.03.21 17:43:32 instance(cloudvideo3): Sending command: FUNCTION CutDirect input=1&mix=0
22.03.21 17:43:32 instance(cloudvideo1): Sending command: FUNCTION CutDirect input=1&mix=0
22.03.21 17:36:18 instance(cloudvideo3): Command Response: VERSION OK 23.0.0.62,
22.03.21 17:36:18 instance(cloudvideo1): Command Response: VERSION OK 24.0.0.62,
22.03.21 17:36:18 instance(cloudvideo3): Connected XML Socket
22.03.21 17:36:18 instance(cloudvideo3): Connected Activator Socket
22.03.21 17:36:18 instance(cloudvideo1): Connected XML Socket
22.03.21 17:36:18 instance(cloudvideo1): Connected Activator Socket
22.03.21 17:36:18 instance(cloudvideo3): Connected Function Socket
22.03.21 17:36:18 instance(cloudvideo1): Connected Function Socket
22.03.21 17:36:18 instance(vmix-asus): Command Response: VERSION OK 25.0.0.24,
22.03.21 17:36:18 instance(vmix-asus): Connected XML Socket
22.03.21 17:36:18 instance(vmix-asus): Connected Activator Socket
22.03.21 17:36:18 instance(vmix-asus): Connected Function Socket

Any help?

Thanks

**Comment by thedist** [comment]
The TCP connection is exactly the same in this version as it has been in previous versions (and is the TCP connection handler that's part of Companion itself), the only difference in the latest update is that in now uses 3 connections per instance to better handle larger vMix productions. So I'm not sure what would be causing your connections to be dropping other than a connection issue between Companion and vMix.

**Comment by nanus** [comment]
Reverted back to v2.1.4 and the problem is not present anymore.
I followed the instructions to prepare the development environment on windows 10 and launched Companion 2.2 with "yarn dev".

Here I attach the log file where I can see some "Unknown TCP message" errors and I noticed that the initial part of the vMix xml response is missing.
At 2022-03-24T09:29:59.162Z there is a button press without any command response and then a ECONNRESET.

How can I provide useful informations to debug what's going on and help troubleshooting what's happening?

Thanks in advance.

[companion_log_20220324-1030.csv](https://github.com/bitfocus/companion-module-studiocoast-vmix/files/8341000/companion_log_20220324-1030.csv)
